src/Cake.SqlServer/SqlDacpacImpl.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+using Microsoft.SqlServer.Dac;
+
+namespace Cake.SqlServer
+{
+    internal static class SqlDacpacImpl
+    {
+        /// <summary>
+        /// Extract the schema from a database into a package.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="targetDatabaseName">The target database.</param>
+        /// <param name="settings">The settings.</param>
+        public static void ExtractDacpacFile(ICakeContext context, string connectionString, string targetDatabaseName, ExtractDacpacSettings settings)
+        {
+            context.Log.Information($"About to extract a dacpac file from database {targetDatabaseName}");
+
+            var service = new DacServices(connectionString);
+
+            service.Extract(settings.OutputFile.FullPath, targetDatabaseName, settings.Name, settings.Version, settings.Description, settings.Tables);
+
+            context.Log.Information($"Finished creating dacpac file from database {targetDatabaseName}. File location is {settings.OutputFile}");
+        }
+
+        /// <summary>
+        /// Extract the schema from a database into a package.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="targetDatabaseName">The target database.</param>
+        /// <param name="dacpacFilePath">Path to the dac file.</param>
+        /// <param name="config">Configure the sql deployment</param>
+        public static void PublishDacpacFile(ICakeContext context, string connectionString, string targetDatabaseName, string dacpacFilePath, Action<PublishOptions> config = null)
```

Comment:
Any specific reason to pass `Action<PublishOptions>` to manipulate freshly created `PublishOptions`? can we not just pass `PublishOptions` directly and do all the configuration before we pass the object into this method? And avoid additional cost of having to pass an `Action<>`?